src/WorldClock.js

```diff
@@ -0,0 +1,110 @@
+import React from "react";
+import Container from "react-bootstrap/Container";
+import Row from "react-bootstrap/Row";
+import Col from "react-bootstrap/Col";
+import "bootstrap/dist/css/bootstrap.min.css";
+import { Form, Button } from "react-bootstrap";
+
+class WorldClock extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      display: <></>,
+    };
+    this.timezonesInput = React.createRef();
+    this.inputForm = (
+      <Form className="w-50" onSubmit={() => this.submit()}>
+        <Form.Group className="mb-3">
+          <Form.Label>Set-up your clocks</Form.Label>
+          <Form.Control
+            type="text"
+            placeholder="Asia/Singapore America/Detroit"
+            ref={this.timezonesInput}
+          />
+        </Form.Group>
+        <Button variant="light" onClick={() => this.submit()}>
+          Show me the time!
+        </Button>
+      </Form>
+    );
+    this.state.display = this.inputForm;
```

Comment:
This whole definition could be simplified I think. Why do you need all the rendered elements in the state? I think it would be wiser to just render those in the component.